dictcc.el

```diff
@@ -271,11 +282,23 @@ At the moment they are of the form `<tr id='trXXX'></tr>'."
                                (dictcc--insert-candidate #'cdr))))))
     (helm :prompt "Filter: " :sources (list source))))
 
+(defun dictcc--select-sourcelang-helm ()
+  "Select source language with helm."
+  (let* ((source `((name . ,"Languages" )
+		   (candidates . ,dictcc-source-languages-alist)
+		   (action . ,(quote identity))
+		   )))
+  (helm :prompt "Filter: " :sources (list source))))
+
+
 ;;;###autoload
 (defun dictcc (query)
   "Search dict.cc for QUERY and insert a result at point."
   (interactive "sQuery: \n")
-  (dictcc--request query))
+  (let ((asklang current-prefix-arg))
+    (dictcc--request query asklang))
```

Comment:
Instead of pushing `asklang` through all these layers, it would be better to use dynamic redefinitions here, e.g.

```emacs-lisp
(let ((dictcc-source-lang (ask-user-for-source-lang-somehow)))
  (dictcc--request query))
```

which would dynamically rebind `dictcc-source-lang` only for the duration of the call to `dictcc--request` without `dictcc--request` having to know about all these shenanigans.